File: src/main/kotlin/ai/astha/scantheplanet/idea/scanner/Scope.kt
package ai.astha.scantheplanet.idea.scanner

import java.nio.file.Path

sealed class ScopeKind {
    object FullRepo : ScopeKind()
    data class File(val file: Path) : ScopeKind()
    data class Selection(val file: Path, val startLine: Int, val endLine: Int) : ScopeKind()
    data class GitDiff(val baseRef: String, val includeUntracked: Boolean) : ScopeKind()
}

class PathFilters(
    val includeExtensions: List<String> = emptyList(),
    val excludeExtensions: List<String> = emptyList(),
    val includeGlobs: List<String> = emptyList(),
    val excludeGlobs: List<String> = emptyList(),
    val maxFileBytes: Long? = null,
    val excludeDocs: Boolean = true,
    val excludeTests: Boolean = true,
    val includeOverride: List<String> = emptyList(),
    val excludePatterns: List<String> = emptyList(),
    val onlyFiles: Set<Path>? = null
)

File: packages/core/src/scanner/Scope.ts
export type ScanScope =
  | { kind: 'full' }
  | { kind: 'file'; file: string }
  | { kind: 'selection'; file: string; startLine: number; endLine: number }
  | { kind: 'gitDiff'; baseRef: string; includeUntracked: boolean };

export interface PathFilters {
  includeExtensions: string[];
  excludeExtensions: string[];
  includeGlobs: string[];
  excludeGlobs: string[];
  maxFileBytes?: number;
  excludeDocs: boolean;
  excludeTests: boolean;
  includeOverride: string[];
  excludePatterns: string[];
  onlyFiles: Set<string> | null;
}
